bff/src/Bff/DynamicFrontends/Internal/ConfigureBffStartupFilter.cs

```diff
@@ -53,15 +58,40 @@ private static void ConfigureOpenIdConfigurationCacheExpiration(IApplicationBuil
         var frontendStore = app.ApplicationServices.GetRequiredService<FrontendCollection>();
         var oidcOptionsMonitor = app.ApplicationServices.GetRequiredService<IOptionsMonitorCache<OpenIdConnectOptions>>();
         var cookieOptionsMonitor = app.ApplicationServices.GetRequiredService<IOptionsMonitorCache<CookieAuthenticationOptions>>();
+        var clientCredentialsCache = app.ApplicationServices.GetRequiredService<IOptionsMonitorCache<ClientCredentialsClient>>();
+        var hybridCache = app.ApplicationServices.GetRequiredKeyedService<HybridCache>(ServiceProviderKeys.ClientCredentialsTokenCache);
+        var logger = app.ApplicationServices.GetRequiredService<ILogger<ConfigureBffStartupFilter>>();
 
         frontendStore.OnFrontendChanged +=
             changedFrontend =>
             {
+                logger.ChangedFrontendDetected_ClearingCaches(LogLevel.Debug, changedFrontend.Name);
```

Comment:
Suggestion, decouple the initial handling of a event from processing the event with a channel. This will allow multiple producers and serialize the processing to one at a time, managing resources, potential concurrent actions on same date (hypothesis) and clean shut down. The channel reader could be in a hosted service.